jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/internal/HttpConnection.java

```diff
@@ -776,7 +778,7 @@ public Action process() throws Exception
                     case HEADER_OVERFLOW:
                     {
                         if (_header.capacity() >= getHttpConfiguration().getResponseHeaderSize())
-                            throw new HttpException.RuntimeException(INTERNAL_SERVER_ERROR_500, "Response header too large");
+                            throw new HttpException.RuntimeException(INTERNAL_SERVER_ERROR_500, "Response Header Fields Too Large");
```

Comment:
Here, the status is (IMHO correctly) set to 500 when headers are too large, unlike H2.

Reply:
Not sure what you mean here? H2 does not generate an HTTP response (it just fails the connection).